library/a-star/src/main/java/astar/AStar.java

```diff
@@ -0,0 +1,259 @@
+package astar;
+
+import static pictureMapper.GetPixelColor.getBoolArr;
+import static pictureMapper.GetPixelColor.getRGBarray;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+
+import javax.imageio.ImageIO;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class AStar {
+
+
+    public static void linkNeighbors(Spot[][] grid) {
+        for (int i = 0; i < grid.length; i++) {
+            for (int j = 0; j < (grid[i].length) - 1; j++) {
+                if (i > 0) {
+                    grid[i][j].addNeighbor(grid[i - 1][j]);
+                }
+                if (j > 0) {
+                    grid[i][j].addNeighbor(grid[i][j - 1]);
+                }
+                if (i < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i + 1][j]);
+                }
+                if (j < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i][j + 1]);
+                }
+                if (i > 0 && j > 0) {
+                    grid[i][j].addNeighbor(grid[i - 1][j - 1]);
+                }
+                if (i < grid.length - 1 && j < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i + 1][j + 1]);
+                }
+            }
+        }
+    }
+
+    public static Map minmaxXY(JSONObject coords, int gridSize, int ratio){
+        // The min max default values are set
+        // respectively to the 25x25 array size
+        // generated by the metadata tool
+        long minX = gridSize;
+        long minY = gridSize;
+        long maxX = 0;
+        long maxY = 0;
+        for(int i = 0; i<coords.size(); i++){
+            JSONObject xy = (JSONObject) coords.get(String.valueOf(i));
+            long x = (long) xy.get("x");
+            long y = (long) xy.get("y");
+
+            if(x<minX){
+                minX = x;
+            }
+            if(x>maxX){
+                maxX = x;
+            }
+            if(y<minY){
+                minY = y;
+            }
+            if(y>maxY){
+                maxY = y;
+            }
+        }
+
+        Map minmax = new Hashtable();
+        minmax.put("minX", minX*ratio);
+        minmax.put("maxX", (maxX*ratio)+ratio);
+        minmax.put("minY", (minY*ratio)-ratio);
+        minmax.put("maxY", maxY*ratio);
+
```

Comment:
I feel like the hash map approach is a very non-oop approach. I dont understand why you would ever want to do this instead of just creating and Object called coordinates. This hashmap approach makes your code harder to understand, as someone who is modifying/adding to your code will have to know the hard-coded keys that you used. if it was an object with ints, they just have to go to the class, but since it is a hashmap, they have to look for where you created the hashmap which is much less obvious

Reply:
I fully agree, after discussion and ponderation, creating a class data type is the way to go for now.